incubator/hnc/internal/validators/object.go

```diff
@@ -164,6 +164,16 @@ func (o *Object) handleInherited(op admissionv1beta1.Operation, newSource, oldSo
 		return deny(metav1.StatusReasonForbidden, "Cannot create objects with the label \""+api.LabelInheritedFrom+"\"")
 
 	case admissionv1beta1.Delete:
+		if o.isDeletingNS(oldInst) {
+			// There are few things more irritating in (K8s) life than having some stupid controller stop
+			// your namespace from being deleted. If there's an object in here and we've decided that the
+			// namespace should be deleted, then don't block anything!
+			//
+			// It's probably slightly safer to create a K8s client, actually load the namespace and check
+			// its deletion timestamp here. But it's much easier to just record this information in the
+			// namespace/HC reconciler and read it here.
+			return allow("allowing deletion of propagated object since namespace is being deleted")
+		}
 		return deny(metav1.StatusReasonForbidden, "Cannot delete object propagated from namespace \""+oldSource+"\"")
```

Comment:
Question: Previously when a user deletes the namespace containing this copy, the object wehbook will deny it with this message? Or it's just the namespace is always terminating and we don't see this message?

Reply:
Yes, that's right - this was the error I was seeing when the e2e tests hung.

Reply:
sg